src/Main.elm

```diff
@@ -37,146 +47,51 @@ init location =
-
-        NavigateToAlerts alertsRoute ->
-            let
-                ( alertsMsg, filter ) =
-                    (Alerts.Update.urlUpdate alertsRoute)
-
-                ( alertGroups, alertCmd ) =
-                    Alerts.Update.update alertsMsg model.alertGroups filter
-            in
-                ( { model | alertGroups = alertGroups, route = AlertsRoute alertsRoute, filter = filter }, Cmd.map alertTranslator alertCmd )
-
-        Alerts alertsMsg ->
-            let
-                ( alertGroups, alertCmd ) =
-                    Alerts.Update.update alertsMsg model.alertGroups model.filter
-            in
-                ( { model | alertGroups = alertGroups }, Cmd.map alertTranslator alertCmd )
-
-        NavigateToSilences silencesRoute ->
-            let
-                ( silencesMsg, filter ) =
-                    (Silences.Update.urlUpdate silencesRoute)
-
-                ( silences, silence, silencesCmd ) =
-                    Silences.Update.update silencesMsg model.silences model.silence filter
-            in
-                ( { model | silence = silence, silences = silences, route = SilencesRoute silencesRoute, filter = filter }
-                , Cmd.map silenceTranslator silencesCmd
-                )
-
-        NavigateToStatus ->
-            ( { model | route = StatusRoute }, getStatus )
-
-        Silences silencesMsg ->
-            let
-                ( silences, silence, silencesCmd ) =
-                    Silences.Update.update silencesMsg model.silences model.silence model.filter
-            in
-                ( { model | silences = silences, silence = silence }
-                , Cmd.map silenceTranslator silencesCmd
-                )
-
-        RedirectAlerts ->
-            ( model, Task.perform NewUrl (Task.succeed "/#/alerts") )
-
-        UpdateFilter filter text ->
-            let
-                t =
-                    if text == "" then
-                        Nothing
-                    else
-                        Just text
-            in
-                ( { model | filter = { filter | text = t } }, Cmd.none )
-
-        NewUrl url ->
-            ( model, Navigation.newUrl url )
-
-        Noop ->
-            ( model, Cmd.none )
-
-        UpdateCurrentTime time ->
-            ( { model | currentTime = time }, Cmd.none )
-
-        MsgForStatus msg ->
-            let
-                ( status, cmd ) =
-                    Status.Update.update msg model.status
-            in
-                ( { model | status = status }, cmd )
-
-
 urlUpdate : Navigation.Location -> Msg
 urlUpdate location =
     let
         route =
             Parsing.urlParser location
     in
         case route of
-            SilencesRoute silencesRoute ->
-                NavigateToSilences silencesRoute
+            SilenceListRoute maybeFilter ->
+                NavigateToSilenceList maybeFilter
+
+            SilenceRoute silenceId ->
+                NavigateToSilence silenceId
```

Comment:
I'll have to check, but we might be able to cut out `NavigateToSilence silenceId` and replace it with `MsgForSilence (InitSilenceView silenceId)`